### ISSUE TITLE
Flaky TRANS_EXTERNAL_02 - balance is off by ~200 Lovelace

### ISSUE THREAD

**Issue by Anviking** [cardano-foundation/cardano-wallet#2287]
# Context
 
 

Separate from #2233 I think.

TRANS_EXTERNAL_02 creates a hard-coded external tx. It verifies that the target wallet eventually has the balance within `faucetAmt - fee_{min, max} - amtSrc`.

# Test Case

https://github.com/input-output-hk/cardano-wallet/blob/2a052d02d9fb8c22bf35daf8c0ceb6af0fa05a55/lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs#L1129-L1140

# Failure / Counter-example

```
Failures:

   src/Test/Integration/Scenario/API/Shelley/Transactions.hs:1030:18:
   1) API Specifications, SHELLEY_TRANSACTIONS, TRANS_EXTERNAL_02 - Multiple Outputs Transaction - Shelley witnesses
        Waited longer than 90s (more than 2 epochs) to resolve action: "wFaucet and wSrc balances are as expected".
        expected: Quantity {getQuantity = 999989843200}
         but got: Quantity {getQuantity = 999989843000}

   To rerun use: --match "/API Specifications/SHELLEY_TRANSACTIONS/TRANS_EXTERNAL_02 - Multiple Outputs Transaction - Shelley witnesses/"

 Randomized with seed 374297169

 Finished in 2152.4762 seconds
 684 examples, 1 failure, 3 pending
```

Off by 200 -> https://github.com/input-output-hk/cardano-wallet/pull/2278#issuecomment-718057319

Some bors trys:
https://github.com/input-output-hk/cardano-wallet/pull/2191#issuecomment-719506298
https://github.com/input-output-hk/cardano-wallet/pull/2191#issuecomment-719926482

---
# Resolution 

---

# QA

**Comment by Anviking** [comment]
When passing locally it seems that it checks that `999989843200 \in [999989843200, 999989843200]`
When failing we have `999989843000 \in [999989843200, 999989843200]`, i.e. the wallet fee estimation is the same, but the wallet balance somehow is 200 less than expected.

minfeeA is `100`, which is half of the discrepancy. I saw `TRANS_CREATE_03` fail locally at one point, where the discrepancy was exactly `1*minfeeA`.

**Comment by Anviking** [comment]
What puzzled me immensely at first was that I thought that the relevant tx here was an externally constructed tx, where the fee shouldn't be variable. But that's not the case. It's the wallet that is constructing it:

https://github.com/input-output-hk/cardano-wallet/blob/d6b7627815ccc592d2664eefc14c3795b2fc1377/lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs#L1107

(mkMultipleTxPayload doesn't do anything related to external txs)

**Comment by Anviking** [comment]
TRANS_CREATE_03 failure I saw a long time ago
```
  src/Test/Integration/Scenario/API/Shelley/Transactions.hs:416:15:
  1) API Specifications, SHELLEY_TRANSACTIONS, TRANS_CREATE_03 - 0 balance after transaction
       expected: Status {statusCode = 202, statusMessage = "Accepted"}
        but got: Status {statusCode = 403, statusMessage = "Forbidden"}

       from the following response: Left (DecodeFailure "Error in $: parsing Cardano.Wallet.Api.Types.ApiTransaction(ApiTransaction) failed, key \"id\" not found: Response {responseStatus = Status {statusCode = 403, statusMessage = \"Forbidden\"}, responseVersion = HTTP/1.1, responseHeaders = [(\"Transfer-Encoding\",\"chunked\"),(\"Date\",\"Mon, 26 Oct 2020 20:34:15 GMT\"),(\"Server\",\"Warp/3.3.5\")], responseBody = \"{\\\"code\\\":\\\"cannot_cover_fee\\\",\\\"message\\\":\\\"I'm unable to adjust the given transaction to cover the associated fee! In order to do so, I'd have to select one or more additional inputs, but I can't do that without increasing the size of the transaction beyond the acceptable limit. Note that I am only missing 4 Lovelace.\\\"}\", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}")

       While verifying (Status {statusCode = 403, statusMessage = "Forbidden"},Left (DecodeFailure "Error in $: parsing Cardano.Wallet.Api.Types.ApiTransaction(ApiTransaction) failed, key \"id\" not found: Response {responseStatus = Status {statusCode = 403, statusMessage = \"Forbidden\"}, responseVersion = HTTP/1.1, responseHeaders = [(\"Transfer-Encoding\",\"chunked\"),(\"Date\",\"Mon, 26 Oct 2020 20:34:15 GMT\"),(\"Server\",\"Warp/3.3.5\")], responseBody = \"{\\\"code\\\":\\\"cannot_cover_fee\\\",\\\"message\\\":\\\"I'm unable to adjust the given transaction to cover the associated fee! In order to do so, I'd have to select one or more additional inputs, but I can't do that without increasing the size of the transaction beyond the acceptable limit. Note that I am only missing 4 Lovelace.\\\"}\", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}"))

  To rerun use: --match "/API Specifications/SHELLEY_TRANSACTIONS/TRANS_CREATE_03 - 0 balance after transaction/"

  src/Test/Integration/Scenario/API/Shelley/Transactions.hs:1037:18:
  2) API Specifications, SHELLEY_TRANSACTIONS, TRANS_EXTERNAL_02 - Multiple Outputs Transaction - Shelley witnesses
       Waited longer than 90s (more than 2 epochs) to resolve action: "wFaucet and wSrc balances are as expected".
       expected: Quantity {getQuantity = 999989842531}
        but got: Quantity {getQuantity = 999989842430}
```

which was actually TRANS_EXTERNAL_02 🤔

**Comment by Anviking** [comment]
Hopefully fixed through relaxed assertions in #2191.
